react/RelatedProducts.js

```diff
@@ -81,8 +89,9 @@ const RelatedProducts = ({
           return null
         }
         const { productRecommendations } = data
+
         const productListProps = {
-          products: productRecommendations || [],
+          products: checkForOutOfStock(productRecommendations),
```

Comment:
I have a suggestion for you, what do you think about it?
```ts
const { productRecommendations = [] } = data

const productListProps = {
  products: hideOutOfStockItems ? filterOutOfStock(productRecommendations) : productRecommendations,
```

Reply:
Or you can do something like that.

```ts
const productListProps = {
  products: hideOutOfStockItems ? filterOutOfStock(productRecommendations) : productRecommendations,
}
```
Inside the filterOutOfStock function
```ts
function filterOutOfStock(products = []) {
   return products.filter(...)
}
```

Reply:
Sure. I've added and tested the changes.